File: TS_JT-Reader/TS_JT-Reader_Vis-Studio/TS_JT-Reader/VecI32.ts
module C_sharp_JT_Reader {
    export class VecI32 {
        private _counter: Int32 = 0;
        private _data: Int32[];
        public setLength(size: number): void {
            this._data = new Array(size);
        }
        public ptr(): number {
            return this._counter++;
        }
    }
}

File: TS_JT-Reader/C#_JT-Reader_Vis-Studio/C sharp JT-Reader/DataType/VecI32.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace C_sharp_JT_Reader
{
    public class VecI32
    {
        private Int32 _counter = 0;
        //private Int32 _count = 0;
        private Int32[] _data;

        // Method to set the length of the array
        public void setLength(int size)
        {
            _data = new Int32[size];
        }

        // Method to reurn a pointer to the next avaible slot in the array
        public int ptr()
        {
            return _counter++;
        }
    }
}
